File: 134-gas-station/gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        start=0
        balance=0
        deficit=0
        for i in range(len(gas)):
            balance+=gas[i]-cost[i]
            if balance<0:
                deficit+=balance
                start=i+1
                balance=0
        if deficit+balance>=0:
            return start
        return -1    

File: 134-gas-station/gas-station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int deficit=0;
        int balance=0;
        int start=0;
        int n=gas.size();
        for(int i=0;i<n;i++)
        {
            balance+=gas[i]-cost[i];
            if(balance<0)
            {
                deficit+=balance;
                start=i+1;
                balance=0;
            }
        }
        if(deficit+balance>=0)
        return start;
        else{
            return -1;
        }
    }
};